mypy_primer.py

```diff
@@ -587,7 +587,7 @@ def select_projects() -> list[Project]:
     if projects == []:
         raise ValueError("No projects selected!")
 
-    if ARGS.num_shards and ARGS.shard_index:
+    if ARGS.num_shards and ARGS.shard_index is not None:
```

Comment:
Hmm, surely `ARGS.shard_index` should never be `None` if `ARGS.num_shards` is not `None`. This is guarded against here:

https://github.com/hauntsaninja/mypy_primer/blob/592a1f7972989e967da17bbbfb851e27aea9202a/mypy_primer.py#L910

Maybe using an `assert` here would be better?

```suggestion
    if ARGS.num_shards:
        assert ARGS.shard_index is not None
```